### ISSUE TITLE
Cannot use any of the features in ZEN

### ISSUE THREAD

**Issue by whiteytidys** [zen-mod/ZEN#622]
**Arma 3 Version:** `x.xx` (stable / rc / dev)
**CBA Version:** `3.x.x` (stable / dev + commit hash)
**ZEN Version:** `1.x.x` (stable / dev + commit hash)

**Mods:**
```
- CBA_A3
- ZEN
```

**Description:**
Add a detailed description of the error. This makes it easier for us to fix the issue.

**Expected behavior:**

**Steps to reproduce:**
Add the steps needed to reproduce the issue.

1. _Go to ..._
2. _Click ..._
3. _See ..._

**Where did the issue occur?**
- Dedicated / Self-Hosted Multiplayer / Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**Additional information:**
Provide any additional information that will help us solve this issue.

**RPT log file:**
- Add a link ([gist](https://gist.github.com) or [pastebin](http://pastebin.com)) to the client and/or server RPT file. An instruction to find your RPT files can be found [here](https://community.bistudio.com/wiki/Crash_Files#Arma_3).

**Screenshots:**

**Comment by whiteytidys** [comment]
Dedicated Server

Mods - 

ace 
ACE Compat - RHS Armed Forces of the Russian Federation 
ACE Compat - RHS United States Armed Forces 
ACE Compat - RHS: GREF 
ACEX 
ADV ACE-CPR (Pike Edition) 
ArmaFXP 
ASR AI3
Blastcore Edited (standalone version) 
BloodLust
Bullet Casings
CBA_A3 
CUP Terrains - Core 
CUP Terrains - Maps 
CUP Units 
CUP Vehicles
CUP Weapons 
Dual Arms - Two Primary Weapons 
Enhanced Movement
Enhanced Soundscape
Females-TCGM_Girls
Flag Interact - Raise and Lower Flags on Flagpoles
Freestyles Crash Landing
Gear Functions
ILBE Assault Pack (TFAR) 
Immersion Cigs
JSRS SOUNDMOD | Steam
JSRS SOUNDMOD - Reloading Sounds
JSRS SOUNDMOD - RHS AFRF Mod Pack Sound 
JSRS SOUNDMOD - RHS GREF Mod Pack Sound 
JSRS SOUNDMOD - RHS USAF Mod Pack Sound 
L3-GPNVG18 Panoramic Night Vision 
LAMBS_Danger.fsm
LAMBS_RPG 
LAMBS_Suppression
Project OPFOR 
Real World Weather 
RHSAFRF
RHSGREF
RHSUSAF 
Specialist Military Arms (SMA) Version 2.7.1 
Task Force Arrowhead Radio
TAW View Distance - Addon
VSM All-In-One Collection

When I use ZEN on my server I can only use the teleport function.  I cannot update edible objects or use any of the other features.

However when I use it on the client side it works fine.

**Comment by mharis001** [comment]
Please fill out the issue template and try with CBA and ZEN only. Also, the mod list you have provided does not include ZEN. ZEN has to be installed on all machines (all clients and the server).

**Comment by whiteytidys** [comment]
Okay, any thoughts on the issue now that I have filled out the form?

**Comment by Kexanone** [comment]
> Okay, any thoughts on the issue now that I have filled out the form?

As @mharis001 has already pointed out, you should reduce your mod list to a minimum for reproducing the error (ideally to CBA and ZEN only).

> However when I use it on the client side it works fine.

What do you mean with that? When you play as server host?

**Comment by whiteytidys** [comment]
When I use ZEN and all my mods via the editor as in loading in and testing features they work.  So it makes me think there is something server side? The mod is loaded in on server side and the keys are uploaded.  If I reduce my mods down to cba and zen only I will break the server.

**Comment by I-Knight-I** [comment]
Downgraded our ZEN version to the release from August 31st 2020 and this bug isn't present. 

Disclaimer though: we've not tested versions after this one. We just recalled using a ZEN version from around then that was fine so went straight for it. So I can't say the exact version it was introduced. Hopefully that's still useful in narrowing it down.

**Comment by whiteytidys** [comment]
Okay, I used it on the server with just CBA and ZEN and it worked, I am using the Hearts and minds game mod, would any of the scripting from that interfere with the Mod?

**Comment by whiteytidys** [comment]
Also on the Game Logic it gives me this message "WARNING: zen requires zen_compat_ace (@ACE) at version [1,11,1,27] (or higher). You Have None."

**Comment by mharis001** [comment]
That message means you need to run the ACE compatibility addon. Since it works now, can this issue be closed?

**Comment by whiteytidys** [comment]
No because with ace combat on it still doesn't work

**Comment by mharis001** [comment]
Issue was fixed by running CBA and ZEN only. Please ensure that ZEN is installed on all machines. Closing as not sure what else we can do with the provided information.

**Comment by I-Knight-I** [comment]
I don't think an adequate solution is "Only use the most fundamental dependency with our mod for it to work". 

> Downgraded our ZEN version to the release from August 31st 2020 and this bug isn't present.

I said that in August. I understand it's not much to go on but if older versions of ZEN work fine, that means there's a regression somewhere.

**Comment by TheRadiumDude** [comment]
> I don't think an adequate solution is "Only use the most fundamental dependency with our mod for it to work".
> 
> > Downgraded our ZEN version to the release from August 31st 2020 and this bug isn't present.
> 
> I said that in August. I understand it's not much to go on but if older versions of ZEN work fine, that means there's a regression somewhere.

This does not necessarily mean it is an issue with ZEN - otherwise we would probably be seeing way more reports of this. It could be that a mod is conflicting with recent ZEN updates, or the server is misconfigured, but neither of these issues imply that it is ZEN's fault since it works very well with many other mods which are frequently updated and commonly used.

That's why we ask for people to see if they can reproduce it with ZEN + CBA_A3, because if they can't, then there is a deeper issue that may not even be related to ZEN and instead related to a very specific combination of mods where new updates may have caused this behaviour. If you can reproduce this concretely *and* identify the mod that is causing the conflict *(and if it is indeed conflicting with ZEN)*, then feel free to open a new issue, if it is indeed a compatibility issue that needs to be fixed our side.

Otherwise, we can't really do much with the information provided.

**Comment by I-Knight-I** [comment]
I respect that it's not much, I really do. 

But I can assure you that - by changing none of our other mods except for downgrading ZEN - we resolved the conflict. 

I'm afraid I don't have the time to test more thoroughly. I last played Arma in August when I made the comment; so I fully understand how you guys can't do much with what I'm providing but the only condition we changed was ZEN's version.

**Comment by whiteytidys** [comment]
Alrighty, here's what I did to correct the issue.

I deleted all the uploaded mods from my server and re-uploaded them one by one with zen to verify it wasn't one of them conflicting.  I imagine one of the mods or the zen mod had been missing a file or was just generally corrupted and wasn't able to work with it.  

Either way I was able to fix it by just wiping the server and starting from scratch which was what I was trying to avoid in the first place.  

Thanks for all your help guys.
